wcomponents-core/src/test/java/com/github/bordertech/wcomponents/servlet/ServletUtilDeviceType_Test.java

```diff
@@ -0,0 +1,77 @@
+package com.github.bordertech.wcomponents.servlet;
+
+import com.github.bordertech.wcomponents.util.DeviceType;
+import com.github.bordertech.wcomponents.util.mock.servlet.MockHttpServletRequest;
+import java.util.Arrays;
+import java.util.Collection;
+import junit.framework.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Parameterized test for device types in {@link ServletUtil}.
+ *
+ * @author Jonathan Austin
+ * @since 1.2.5
+ */
+@RunWith(Parameterized.class)
+public class ServletUtilDeviceType_Test {

```

Comment:
I like the tests. 

Do you think we should add UC user agent string to the list?

Also some of those user agent strings look like templates, for example `<Build Tag etc.>` do you think we should actually file them out or turn them into sprintf strings?